src/api/routes/edition-group.js

```diff
@@ -228,10 +254,16 @@ router.get('/',
 
 		if (req.query.modelType === 'Edition') {
 			// If we're loading an Edition, also load the related Edition Group from the ORM model
+			const {Edition} = req.app.locals.orm;
+			const {bbid} = req.query;
 			const relationships = editionGroupBasicRelations.map(rel => `editionGroup.${rel}`);
-			await makeEntityLoader(null, relationships, 'Entity not found', true, false)(req, res, next);
-			const {editionGroup} = res.locals.entity;
-			// edition will belong to only one edition-group
+			const edition = await new Edition({bbid}).fetch({
+				require: false,
+				withRelated: relationships
+			});
+			const editionJSON = edition ? edition.toJSON() : {};
```

Comment:
Here it looks like you're loading the Edition twice: once in the `makeEntityLoader` middleware definde on the route and once manually here.
This `await new Edition({bbid})` could be remove, as res.locals.entity already has the entity. What it doesn't have is the related EditionGroup (the extra editionGroup.XXX relationships defined a line above), which could be added to the `makeEntityLoader` call.
In this case for EditionGroups that can only have relationships to Editions its easy enough:
```makeEntityLoader(null, utils.relationshipsRelations.concat(editionGroupBasicRelations.map(rel => `editionGroup.${rel}`)), 'Entity not found', true),```

However to make it work for other entities, you could add a new flag to `makeEntityLoader` which, when set to true, would also load related `relations.map(rel => ${_.camelCase(modelType)}.${rel})`.
That might work.

If you don't want to add a flag, you could also check if `req.baseUrl`, `isBrowse`  and `req.query.modelType` and make a decision based on that:
For example, if `isBrowse` is true and baseUrl ends with `/edition-group` and `modelType` is Edition, then also load those extra relationships `relations.map(rel => ${_.camelCase(modelType)}.${rel})`. Seems more convoluted to me and moves some logic in makeEntityLoader instead of a simple flag on the browse route, but it's doable if a new flag is an issue.

Reply:
I ended up solving this with a custom middleware in the browse routes definitions that calls makeEntityLoader with the extra related elements to avoid fetching again in the route's main body.

There is one that cannot be optimised currently: `/edition?publisher=…`, since Publisher().editions() !== Publisher.related('editions') (https://github.com/bookbrainz/bookbrainz-data-js/blob/master/src/models/data/publisherData.js#L36).
This could possibly be refactored to use [hasMany().through()](https://bookshelfjs.org/api.html#Model-instance-through) to allow fetching with the Publisher.fetch({withRelated: 'editions.xyz'}) format.